src/app/components/wrapper-components/part-creation/part-creation.component.spec.ts

```diff
@@ -9,620 +12,715 @@ import { PartDAO } from 'src/app/dao/PartDAO';
+                void partDAO.set('joinerId', PartMocks.INITIAL);
 
                 // When arriving on that component
-                testUtils.detectChanges();
-                tick(3000); // needs to be >2999
+                tick();
 
                 // Then the part and all its related data should be removed
-                expect(gameService.deletePart).toHaveBeenCalledWith('joinerId');
-                expect(joinerService.deleteJoiner).toHaveBeenCalledWith();
-                expect(chatService.deleteChat).toHaveBeenCalledWith('joinerId');
+                expect(gameService.deletePart).not.toHaveBeenCalled();
+                expect(joinerService.deleteJoiner).not.toHaveBeenCalled();
+                expect(chatService.deleteChat).not.toHaveBeenCalled();
+                component.stopSendingPresenceTokensAndObservingUsersIfNeeded();
             }));
         });
-        describe('Waiting for config', () => {
-            it('should reroute to server when game is cancelled', fakeAsync(async() => {
+        describe('Leaving', () => {
+            it('should remove yourself when leaving the room', fakeAsync(() => {
+                // Given a partCreation where user is but candidate
+                awaitComponentInitialisation();
+                expect(component.currentJoiner).toEqual(JoinerMocks.WITH_FIRST_CANDIDATE);
+
+                // When leaving the page (testing here by calling ngOnDestroy)
+                spyOn(joinerService, 'unsubscribe').and.callFake(() => {});
+                spyOn(joinerService, 'cancelJoining').and.callFake(async(user: MinimalUser) => {});
+                void component.ngOnDestroy();
+                tick();
+
+                // Then user's name taken out of the candidate list
+                expect(joinerService.cancelJoining).toHaveBeenCalledOnceWith(UserMocks.OPPONENT_MINIMAL_USER);
+            }));
+            it('should remove observedPart when leaving the room', fakeAsync(() => {
+                // Given a partCreation where user is but candidate
+                awaitComponentInitialisation();
+                expect(component.currentJoiner).toEqual(JoinerMocks.WITH_FIRST_CANDIDATE);
+
+                // When leaving the page (testing here by calling ngOnDestroy)
+                const authService: ConnectedUserService = TestBed.inject(ConnectedUserService);
+                spyOn(authService, 'removeObservedPart').and.callThrough();
+                spyOn(joinerService, 'unsubscribe').and.callFake(() => {});
+                spyOn(joinerService, 'cancelJoining').and.callFake(async(user: MinimalUser) => {});
+                void component.ngOnDestroy();
+                tick();
+
+                // Then observed part should be removed
+                expect(authService.removeObservedPart).toHaveBeenCalledOnceWith();
+            }));
+        });
+        describe('Hoping to get chosen', () => {
```

Comment:
aka, 'candidate'? :D

Reply:
oui mais non, le chosenOpponent est candidate aussi !

Reply:
mais les tests là sont pas spécifiques à un candidate qui n'est pas chosenOpponent il semble ?

Reply:
non en effet, là "hoping to get chosen" est en effet un peu neutre comme sous describe du describe('candidate') mais si tu as une meilleurs idée que candidate > candidate suggère :p